src/Core/String/CharacterCleaner.php

```diff
@@ -31,16 +31,16 @@ class CharacterCleaner
     /**
      * Delete unicode class from regular expression patterns.
      *
+     * @deprecated since version 8.0.0, to be removed.
+     *
      * @param string $pattern
      *
      * @return string pattern
      */
     public function cleanNonUnicodeSupport($pattern)
     {
-        if (!defined('PREG_BAD_UTF8_OFFSET')) {
-            return $pattern;
-        }
+        \Tools::displayAsDeprecated('The use of cleanNonUnicodeSupport is not required');
```

Comment:
Please do not use this call, follow the coding standard instead:
https://devdocs.prestashop.com/1.7/development/coding-standards/#deprecations